src/main/scala/dpla/ingestion3/mappers/providers/OhioMapping.scala

```diff
@@ -114,13 +116,15 @@ class OhioMapping extends Mapping[NodeSeq] with XmlExtractor with IdMinter[NodeS
   // OreAggregation
   override def dplaUri(data: Document[NodeSeq]): URI = mintDplaItemUri(data)
 
-  override def dataProvider(data: Document[NodeSeq]): EdmAgent = {
+  override def dataProvider(data: Document[NodeSeq])
+                           (implicit msgCollector: MessageCollector[IngestMessage]): EdmAgent = {
     extractStrings(data \ "metadata" \\ "dataProvider")
       .map(nameOnlyAgent)
-      .headOption // take the first value
-      .getOrElse( // return the first value or throw an exception
-      throw new Exception(s"Missing required property metadata/dataProvider is empty for ${getProviderId(data)}")
-    )
+      .headOption match {
+        case Some(dp) => dp
+        case None => msgCollector.add(missingRequiredError(getProviderId(data), "dataProvider"))
+          nameOnlyAgent("")  // FIXME this shouldn't have to return an empty value.
```

Comment:
Dumb question: Should this just throw if we can't assign a data provider? Or I guess, in the new world, log a fatal message?

Reply:
Yeah, that is what 125 is doing. `errors` are treated as fatal and I chose not to raise an exception because that would end the collection of messages for a record and I was striving to collect all messages (if a mapper has implemented them).

Reply:
Yeah I think I understand the problem a lot better now that we talked.